stytch/user.go

```diff
@@ -100,6 +112,205 @@ type UsersDeleteWebAuthnRegistrationResponse struct {
 	UserID     string `json:"user_id,omitempty"`
 }
 
+/* User Search */
+type UsersSearchParams struct {
+	Limit  int32            `json:"limit,omitempty"`
+	Query  UsersSearchQuery `json:"query,omitempty"`
+	Cursor string           `json:"cursor,omitempty"`
+}
+
+type UsersSearchOperator string
+
+const (
+	UserSearchOperatorOR  UsersSearchOperator = "OR"
+	UserSearchOperatorAND UsersSearchOperator = "AND"
+)
+
+func marshalFilter(filterName string, filterValue interface{}) ([]byte, error) {
+	return json.Marshal(map[string]interface{}{
+		"filter_name":  filterName,
+		"filter_value": filterValue,
+	})
+}
+
+type UsersSearchQuery struct {
+	Operator UsersSearchOperator `json:"operator,omitempty"`
+	Operands []json.Marshaler    `json:"operands,omitempty"`
+}
+
+/* Created At Filters */
+type UsersSearchQueryCreatedAtGreaterThanFilter struct {
+	CreatedAtGreaterThan time.Time
+}
+
+func (q UsersSearchQueryCreatedAtGreaterThanFilter) MarshalJSON() ([]byte, error) {
+	return marshalFilter("created_at_greater_than", q.CreatedAtGreaterThan)
+}
+
+type UsersSearchQueryCreatedAtLessThanFilter struct {
+	CreatedAtLessThan time.Time
+}
+
+func (q UsersSearchQueryCreatedAtLessThanFilter) MarshalJSON() ([]byte, error) {
+	return marshalFilter("created_at_less_than", q.CreatedAtLessThan)
+}
+
+type UsersSearchQueryCreatedAtBetweenFilter struct {
+	CreatedAtGreaterThan time.Time
+	CreatedAtLessThan    time.Time
```

Comment:
```suggestion
	GreaterThan time.Time
	LessThan    time.Time
```

These two shouldn't have the `CreatedAt` prepended.

Reply:
https://github.com/stytchauth/stytch-go/pull/46/commits/5c03693af1451ef227073d5d89559d64a5e8f0d0